src/message_buffer.cpp

```diff
@@ -80,128 +87,192 @@ MessageBuffer::operator=(MessageBuffer&& other)
   _buffer = std::move(other._buffer);
 }
 
+#ifndef htonll
+#define htonll(x) ((static_cast<uint64_t>(htonl(x)) << 32) + htonl((x) >> 32))
```

Comment:
Given this assumes the machine is little-endian, this would do the same and save two API calls:
```cpp
#define htonll(x) ((((x) >> 56) & 0x00000000000000ff) | \
                   (((x) >> 40) & 0x000000000000ff00) | \
                   (((x) >> 24) & 0x0000000000ff0000) | \
                   (((x) >>  8) & 0x00000000ff000000) | \
                   (((x) <<  8) & 0x000000ff00000000) | \
                   (((x) << 24) & 0x0000ff0000000000) | \
                   (((x) << 40) & 0x00ff000000000000) | \
                   (((x) << 56) & 0xff00000000000000))
```

Reply:
Adding to this, do we want this function to take endianness into account?  All we need to do is return this if LE and return the original value if BE.

Reply:
To answer the second comment, where this is used has a constexpr check for endianess, this one isn't doing that since it is a macro (though maybe if htonll macro isn't defined then we could define it as a function)